src/main/java/lotto/domain/ProfitCalculator.java

```diff
@@ -0,0 +1,34 @@
+package lotto.domain;
+
+import lotto.constant.LottoPrize;
+
+import java.util.Map;
+
+public class ProfitCalculator {
+    private final int purchaseAmount;
+    private final Map<LottoPrize, Integer> winningResults;
+
+    public ProfitCalculator(int purchaseAmount, Map<LottoPrize, Integer> winningResults) {
+        this.purchaseAmount = purchaseAmount;
+        this.winningResults = winningResults;
+    }
+
+    public long calculateTotalProfit() {
```

Comment:
값을 감싼 객체로 반환해도 좋을 것 같아요
```suggestion
    public TotalProfit calculateTotalProfit() {
```